### PR TITLE
add lsp-clangd.el

### PR DESCRIPTION
**Pull request by elxbarbosa**
Changes:

- Extract C/CPP language servers to lsp-c-languages.el (https://github.com/emacs-lsp/lsp-mode/pull/2083/commits/d1a42f2c29d007ce3be1a6f04ef69742826e0aa6)
- C language tests unified  
- move and rename lsp-join-region near to lsp-clangd (https://github.com/emacs-lsp/lsp-mode/pull/2083/commits/9f66e984e59cb46098fed8fcac7c49f5279cf959)

### CODE REVIEW

**Comment by ericdallo** [comment]
This one is failing on CI @elxbarbosa

**Comment by elxbarbosa** [comment]
I don't have the slightest clue but let me try :)

**Comment by elxbarbosa** [comment]
Oh of course, the test do not know where that function is( lsp-clangd.el)

**Comment by elxbarbosa** [comment]
Talking about clangd, I can't find where ccls server files are. There is information about it everywhere.

**Comment by leungbk** [comment]
> I can't find where **ccls** definitions are. there is information about it scattered all over the project but no sign of it.

ccls is in an external repo: https://github.com/MaskRay/emacs-ccls.git

**Comment by elxbarbosa** [comment]
@leungbk 

Indeed.

Although its instructions its greatly detailed, ends users may not be that entitle to do all that :)

**Comment by elxbarbosa** [comment]
renaming ccls to lsp-ccls may be improved readability and searchability: 
https://github.com/emacs-lsp/lsp-mode/blob/70551a71b5d3f98c0b4d4b113107db9029617caa/lsp-mode.el#L351

File: test/lsp-clangd-test.el [hunk 1]
```diff
@@ -15,10 +15,14 @@
...
+
+;; Tests for the Clangd  Language Server.
+
 ;;; Code:
 
 (require 'ert)
-(require 'lsp-clients)
+(require 'lsp-clangd)
 
 (ert-deftest lsp-clients-extract-signature-from-clangd-on-hover ()
```
Comment (ericdallo) [inline test/lsp-clangd-test.el:27]:
```suggestion
(ert-deftest lsp-clangd-extract-signature-on-hover ()
```

File: test/lsp-clangd-test.el [hunk 2]
```diff
@@ -58,4 +62,4 @@
                   int k);")
```
Comment (ericdallo) [inline test/lsp-clangd-test.el:62]: We should also rename this test to something like: `lsp-clangd-join-region`

**Comment by kiennq** [comment]
we have `lsp-cpp`, can this two be merged?

File: test/lsp-clangd-test.el (hunk 1 above)
Comment (elxbarbosa) [reply test/lsp-clangd-test.el:27]: done!

**Comment by elxbarbosa** [comment]
@kiennq 
Indeed.!

File: test/lsp-clangd-test.el (hunk 2 above)
Comment (elxbarbosa) [reply test/lsp-clangd-test.el:62]: aint this test meant for usage in all the project?

Comment (ericdallo) [reply test/lsp-clangd-test.el:62]: Yeah, but it's used only on clang, or you rename it for clangd or move it to a lsp-test more generic or something like this

Comment (elxbarbosa) [reply test/lsp-clangd-test.el:62]: fair

Comment (elxbarbosa) [reply test/lsp-clangd-test.el:75]: done!

**Comment by elxbarbosa** [comment]
> we have `lsp-cpp`, can this two be merged?

Indeed. Doing it in a separate PR so as to not confuse whoever read changes later!

**Comment by elxbarbosa** [comment]
@MaskRay @yyoncho 
hey, Ive changed from ccls to lsp-ccls but I aint aware on how it breaks, if it will,  emacs-ccls

**Comment by elxbarbosa** [comment]
@ericdallo 
Undoing renaming ccls. Better you all decide about that :)

**Comment by ericdallo** [comment]
Yeah, IMO we can just move the file and keep the names until we decide the next name.

**Comment by elxbarbosa** [comment]
You mean keep clangd.el instead of c-languages ?

**Comment by ericdallo** [comment]
Yes

**Comment by elxbarbosa** [comment]
@ericdallo 
Done https://github.com/emacs-lsp/lsp-mode/pull/2083/commits/5bb820d35e4c3228a75184ef972079fd74de986b

**Pull request merged**